### ISSUE TITLE
Is it possible to perform a series of actions with a set of elements?

### ISSUE THREAD

**Issue by pzatrick** [webdriverio/webdriverio#941]
Hi,

I really want to be able to use webdriverio, nice API :+1: The thing that's holding me back is the ability to do something like this:

(note: this is pseudocode)

```javascript
webdriverio
  .remote(options)
  .init()
  .elements("button.delete-button").then(function (buttons) {
    buttons.value.forEach(function (button) {
      this.click(button);
      this.click("button.confirm");
    });
  });
```

Is it already possible to achieve the same effect with different code? If not, is something like this on the roadmap?

**Comment by garrickajo** [comment]
See https://github.com/webdriverio/webdriverio/issues/273

**Comment by christian-bromann** [comment]
@garrickajo thanks for pointing out that example.

@pzatrick all commands in WebdriverIO are promises, so to do actions like this you might need the help of some promise libraries or do something like:

```js
var clicks;
var runner = webdriverjs.remote(options);

runner
    .init()
    .url("http://www.google.com")
     // fetch elements
    .elements('a').then(function(res){
        // iterate through elements
        clicks = (clicks || runner).elementIdClick(elem.ELEMENT));
        return clicks;
    })
```

However this is only for commands where you interact with elements. If you grep information from elements and your selector queries multiple elements you will receive an array of results like:

```js
runner.getAttribute('a', 'href').then(function(attrs) {
    console.log(attrs); // array of all href urls of all links on the page
});
```

If you have further questions join the gitter chat and let me know :wink:

**Comment by pzatrick** [comment]
Thanks for the help everyone!

On Tuesday, December 15, 2015, Christian Bromann <notifications@github.com>
wrote:

> Closed #941 <https://github.com/webdriverio/webdriverio/issues/941>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webdriverio/webdriverio/issues/941#event-492901877>.
>


-- 

A gentleman is one who puts more into the world thank he takes out.

-- *GB Shaw*
patrickcanfield.com
